aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Merger.java

```diff
@@ -128,63 +126,35 @@
     }
 
     // Delete a model.
-    private <T extends Model> Completable delete(T model, Consumer<StorageItemChange<T>> onStorageItemChange) {
-        return Completable.defer(() -> Completable.create(emitter -> {
-            // First, check if the thing exists.
-            // If we don't, we'll get an exception saying basically,
-            // "failed to delete a non-existing thing."
-            ifPresent(model,
-                () -> localStorageAdapter.delete(
-                    model,
-                    StorageItemChange.Initiator.SYNC_ENGINE,
-                    QueryPredicates.all(),
-                    storageItemChange -> {
-                        onStorageItemChange.accept(storageItemChange);
-                        emitter.onComplete();
-                    },
-                    emitter::onError
-                ),
-                emitter::onComplete
-            );
-        }));
+    private <T extends Model> Completable delete(T model, Consumer<StorageItemChange.Type> changeTypeConsumer) {
+        return Completable.create(emitter ->
+            localStorageAdapter.delete(model, StorageItemChange.Initiator.SYNC_ENGINE, QueryPredicates.all(),
+                storageItemChange -> {
+                    changeTypeConsumer.accept(storageItemChange.type());
+                    emitter.onComplete();
+                },
+                failure -> {
+                    LOG.verbose(
```

Comment:
This PR has two main changes:
1. Remove the `ifPresent(...)` check;
2. Only log this `failure`, instead of calling `emitter.onError(...)`.